### ISSUE TITLE
[release/2.12][PVC] hf_GPT2_large amp_fp16/float16 inference got eager_two_runs_differ

### ISSUE THREAD

**Issue by mengfei25** [intel/torch-xpu-ops#3464]
### 🐛 Describe the bug

torchbench hf_GPT2_large got `eager_two_runs_differ` with amp_fp16 and float16 inference on PVC with PyTorch 2.12 RC wheel. This means two consecutive eager-mode runs produce different outputs, indicating non-determinism in the model execution on XPU.

This is different from #3058, which tracks `fail_accuracy` in **training** mode (nan in RMSE). This issue is about **inference-mode** non-determinism in eager execution.

Accuracy verification confirmed this is a reproducible failure.

### Error Log

```
loading model: 0it [00:11, ?it/s]
xpu  eval  hf_GPT2_large
E0424 05:45:03.925000 195433 torch/_dynamo/utils.py:3535] Accuracy failed: allclose not within tol=0
E0424 05:45:03.926000 195433 torch/_dynamo/utils.py:3357] Accuracy failed for key name logits
eager_two_runs_differ

```

Torch commit: 489deee763382be6f9ccb296797b782d7054a06a
Triton: 3.7.1

### Reproducer

```bash
docker run -it -e TZ=Asia/Shanghai --device=/dev/mem --device=/dev/dri \
  --group-add video --privileged --shm-size=8g intelgpu/ubuntu-24.04-lts2:2523.40 bash

curl -LsSf https://astral.sh/uv/install.sh | env UV_INSTALL_DIR="/usr/local/bin" sh
uv venv myvenv --python 3.12 --clear
source myvenv/bin/activate
uv pip install pip numpy 'setuptools<81' wheel

uv pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/test/xpu

pytorch_commit="$(python -c 'import torch; print(torch.version.git_version)')"
git clone https://github.com/pytorch/pytorch && cd pytorch && git checkout ${pytorch_commit}
rm -rf torch torch-xpu-ops
git clone https://github.com/intel/torch-xpu-ops
rsync -avz ./torch-xpu-ops/.ci/benchmarks/ ./benchmarks/dynamo/

uv pip install pandas psutil scipy requests
uv pip install -U numpy==1.26.4

git clone https://github.com/pytorch/benchmark tmp-benchmark && cd tmp-benchmark
uv pip install -r requirements.txt && python install.py hf_GPT2_large && cd ..

# Reproduce amp_fp16:
python ./benchmarks/dynamo/torchbench.py --accuracy --amp --amp-dtype float16 \
  --inference -d xpu -n 10 --only hf_GPT2_large --backend=inductor \
  --cold-start-latency --timeout 10800 --disable-cudagraphs

# Reproduce float16:
python ./benchmarks/dynamo/torchbench.py --accuracy --float16 \
  --inference -d xpu -n 10 --only hf_GPT2_large --backend=inductor \
  --cold-start-latency --timeout 10800 --disable-cudagraphs
```

### Analysis

- `eager_two_runs_differ` means the dynamo accuracy checker runs the model twice in eager mode and gets different outputs, so it cannot establish a reliable baseline for accuracy comparison.
- This affects only fp16-related dtypes (amp_fp16, float16), not bf16 or float32, suggesting a numeric instability in float16 eager execution path.
- Related: #3058 tracks training-mode `fail_accuracy` (nan RMSE), which has a different root cause (unfuse_bias_add_to_pointwise dtype guard issue).

### Versions

- PyTorch 2.12 RC (torch commit: 489deee763382be6f9ccb296797b782d7054a06a)
- Triton: 3.7.1
- Platform: PVC (Intel Data Center GPU Max)
- Docker: intelgpu/ubuntu-24.04-lts2:2523.40

**Comment by chuanqi129** [comment]
Last good commit: https://github.com/pytorch/pytorch/commit/f9cca30e8ffdb958c546ef95f0fd7f95685bccc1
Bad commit: https://github.com/pytorch/pytorch/commit/c90ce2bf4a36c337c2d35814d55c7524aa484396

**Comment by weishi-deng** [comment]
on my local pvc 1100, the release 2.12 rc passed with no err. @mengfei25, which PVC do you use?

**Comment by weishi-deng** [comment]
gemm issue in MFDNN-14998

**Comment by weishi-deng** [comment]
@chuanqi129 pls assign someone to check the latest status for this model. OneDNN team says they cannot reproduce the issue from the verbose logs on main, or it could be already fixed by https://github.com/uxlfoundation/oneDNN/pull/5316.

**Comment by weishi-deng** [comment]
mengfei pls check the status after oneDNN update.

**Comment by bjarzemb** [comment]
I have `eager_two_runs_differ` for float16 during pre RC test with torch 2.14.0.dev20260805+xpu on BMG B580 Windows. Test passed in PT2.13

**Comment by mengfei25** [comment]
> mengfei pls check the status after oneDNN update.

Passed on PVC and BMG Linux with release 2.13 and 2.14.0.dev20260805+xpu

**Comment by mengfei25** [comment]
> I have `eager_two_runs_differ` for float16 during pre RC test with torch 2.14.0.dev20260805+xpu on BMG B580 Windows. Test passed in PT2.13

@bjarzemb  Looks like only Windows has such issue now, not same as Linux, please double check and create another issue to track it if does be failed

**Comment by bjarzemb** [comment]
Windows issue created: https://github.com/intel/torch-xpu-ops/issues/4984. Closing this issue.
